src/Common/Core/Impl/Threading/IMainThread.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.Common.Core.Threading {
+    public interface IMainThread {
```

Comment:
That's a good question. Core is referenced by Broker, which has no access to the UI and to concept of Main Thread. Host.Client also should have no idea about UI. We may eventually have to add assembly, something like `Common.Wpf` or `Common.UI` or `Core.UI` to store such things, and also move several methods from ICoreShell there as well.

---
In reply to: [81221364](https://github.com/Microsoft/RTVS/pull/2408#discussion_r81221364) [](ancestors = 81221364)